### PR TITLE
Create labeler.yml

### PR DESCRIPTION
**Pull request by jolorunyomi**
# Summary
This PR adds a GitHub action (PR Labeler) to auto label PRs based on a labeling rule and changes in PR.

### CODE REVIEW

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]
Please see the similar comment on the cuDF PR below about updating the file/path patterns.

https://github.com/rapidsai/cudf/pull/7044#discussion_r550248866

**Review by ajschmidt8 — APPROVED** [review]

File: .github/labeler.yml
Comment (afender) [inline .github/labeler.yml:17]: It should be fine, but just to confirm we do not have a `java` folder yet.

File: .github/labeler.yml
Comment (afender) [inline .github/labeler.yml:7]: @rlratzel should the `benchmark` directory be there?

**Review by afender — COMMENTED** [review]
I noticed there's no labels for :
- `docs`
- `datasets`
- `md` files

This would be something to consider adding to make sure labels are always auto-created.

File: .github/labeler.yml
Comment (rlratzel) [reply .github/labeler.yml:7]: yes, the benchmark dir is python code we run nightly and as part of CI (in test-only mode) so we'd want that added as well.

File: .github/labeler.yml
Comment (rlratzel) [inline .github/labeler.yml:21]: I'm assuming this is boilerplate we don't need to review, but the leading `/` here for the `ci` folder looked suspicious.

**Comment by afender** [comment]
tangential topic: do we have doc owners for review (ie. owners for `md` and `rst` files)? I haven't seen that but we probably should as doc is expected to get upgraded a lot in the near future.

File: .github/labeler.yml
Comment (ajschmidt8) [inline .github/labeler.yml:2]:
```suggestion
# Adapted from https://github.com/rapidsai/cugraph/blob/main/.github/CODEOWNERS
```

should this be `cugraph` instead of `cudf`?

**Review by ajschmidt8 — CHANGES_REQUESTED** [review]
one minor change requested. also, can you include a description on the PR?

**Review by rlratzel — APPROVED** [review]
Approving assuming AJ's question gets resolved and a description of the PR is added.

File: .github/labeler.yml
Comment (jolorunyomi) [reply .github/labeler.yml:2]: Cugraph doesn't have a code owners file. So it's actually adapted from the code owners in `cudf`.

**Review by ajschmidt8 — APPROVED** [review]

**Pull request merged**
